Sources/Core/SharedModels/Assets/OnLedgerEntity.swift

```diff
@@ -5,36 +5,132 @@ import Prelude
 // MARK: - OnLedgerEntity
 public enum OnLedgerEntity: Sendable, Hashable, Codable {
 	case resource(Resource)
-
-	public var address: Address {
-		switch self {
-		case let .resource(resource):
-			return resource.resourceAddress.asGeneral()
-		}
-	}
+	case account(Account)
+	case resourcePool(ResourcePool)
+	case validator(Validator)
+	case nonFungibleToken(NonFungibleToken)
+	case accountNonFungibleIds(AccountNonFungibleIdsPage)
+	case genericComponent(GenericComponent)
 
 	public var resource: Resource? {
 		guard case let .resource(resource) = self else {
 			return nil
 		}
 		return resource
 	}
+
+	public var nonFungibleToken: NonFungibleToken? {
+		guard case let .nonFungibleToken(nonFungibleToken) = self else {
+			return nil
+		}
+		return nonFungibleToken
+	}
+
+	public var accountNonFungibleIds: AccountNonFungibleIdsPage? {
+		guard case let .accountNonFungibleIds(ids) = self else {
+			return nil
+		}
+		return ids
+	}
+
+	public var account: Account? {
+		guard case let .account(account) = self else {
+			return nil
+		}
+		return account
+	}
+
+	public var resourcePool: ResourcePool? {
+		guard case let .resourcePool(resourcePool) = self else {
+			return nil
+		}
+		return resourcePool
+	}
+
+	public var validator: Validator? {
+		guard case let .validator(validator) = self else {
+			return nil
+		}
+		return validator
+	}
+
+	public var genericComponent: GenericComponent? {
+		guard case let .genericComponent(genericComponent) = self else {
+			return nil
+		}
+		return genericComponent
+	}
 }
 
 // MARK: OnLedgerEntity.Resource
 extension OnLedgerEntity {
-	public struct Resource: Sendable, Hashable, Codable, Identifiable {
-		public var id: ResourceAddress { resourceAddress }
-		public let resourceAddress: ResourceAddress
-		public let divisibility: Int?
+	public struct Metadata: Sendable, Hashable, Codable {
+		public enum PublicKeyHash: Sendable, Hashable, Codable {
+			case ecdsaSecp256k1(String)
```

Comment:
I'd prefer it if the associated value type was not `String`, either `Data` or even better: `HexCodable32Bytes`.

Reply:
Not sure if PublickKeyHash is 32 bytes exactly, will preserve String for now, as per how it is defined, and check on this again when we will actually use `manifestForAuthKeyCreation`